src/lib/InstantSearch.js

```diff
@@ -298,15 +302,15 @@ See: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-o
       this._onHistoryChange = noop;
     }
 
-    this.searchParameters = this.widgets.reduce(
+    const initialSearchParameters = this.widgets.reduce(
       enhanceConfiguration,
-      this.searchParameters
+      this._searchParameters
     );
 
     const helper = algoliasearchHelper(
       this.client,
-      this.searchParameters.index || this.indexName,
-      this.searchParameters
+      initialSearchParameters.index || this.indexName,
+      initialSearchParameters
```

Comment:
I'm not sure from this code why this is initialSearchParameters, instead of `this.helper.state`, can a comment be put for the reason?

Reply:
We use `initialSearchParameters` to create the `helper`, we can't use `helper.state` to create it. Which part do you think is unclear? Note that almost nothing changed from the previous implementation.

Reply:
I just mean that in the earlier parts of the code you managed to replace it with `this.helper`

Reply:
Yes but inside the others parts of the code the `helper` is already created. Here it's not the case, we're inside the `start` method the `helper` does not exist yet. I could have inlined the returned value of the `reduce` to avoid the extra variable but we have to branch on the `indexName`.